sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -878,9 +878,10 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
     new Exception(s"Unsupported type: ${dataType.catalogString}")
   }
 
-  def useDictionaryEncodingWhenDictionaryOverflowError(): Throwable = {
-    new IllegalStateException(
```

Comment:
Is this a user-facing exception? or internal exception that's not exposed to end usrs?

Reply:
This appears to be an internal exception not exposed to users

Reply:
@HyukjinKwon  This appears to be an internal exception not exposed to users